src/crl/interactivesessions/_process.py

```diff
@@ -249,8 +250,17 @@ class _BackgroundProcessWithoutPty(_BackgroundProcessBase,
 class _NoCommBackgroudProcess(_BackgroundProcessWithoutPty):
 
     def run(self):
-        self._initialize_terminal()
-        return self._get_cmd_pid()
+        with self._in_terminal():
+            self._initialize_terminal()
+            return self._get_cmd_pid()
+
+    @contextmanager
+    def _in_terminal(self):
+        try:
+            yield None
+        finally:
+            if self.terminal is not None:
+                self.terminalpools.put(self.terminal)
```

Comment:
Could use self._finalize_terminal() to match self._initialize_terminal()

Reply:
Done